### ISSUE TITLE
3.0.0RC1 PHAR does not work with PHP 7.1

### ISSUE THREAD

**Issue by sebastianbergmann** [squizlabs/PHP_CodeSniffer#1166]
```
$ php --version
PHP 7.1.0RC2 (cli) (built: Sep 15 2016 07:21:57) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.1.0-dev, Copyright (c) 1998-2016 Zend Technologies
    with Xdebug v2.5.0-dev, Copyright (c) 2002-2016, by Derick Rethans
```

```
$ wget https://github.com/squizlabs/PHP_CodeSniffer/releases/download/3.0.0RC1/phpcs.phar
```

```
$ php phpcs.phar 
PHP Fatal error:  Uncaught Error: [] operator not supported for strings in phar:///home/sb/phpcs.phar/src/Ruleset.php:2
Stack trace:
#0 phar:///home/sb/phpcs.phar/src/Runner.php(2): PHP_CodeSniffer\Ruleset->__construct(Object(PHP_CodeSniffer\Config))
#1 phar:///home/sb/phpcs.phar/src/Runner.php(2): PHP_CodeSniffer\Runner->init()
#2 /home/sb/phpcs.phar(6): PHP_CodeSniffer\Runner->runPHPCS()
#3 {main}
  thrown in phar:///home/sb/phpcs.phar/src/Ruleset.php on line 2
```

**Comment by aik099** [comment]
@sebastianbergmann , does PHAR of last 2.x version work on PHP 7?

**Comment by gsherwood** [comment]
Not a phar-specific issue, and not in 2.x. I found and fixed this on the 3.0 branch after release: https://github.com/squizlabs/PHP_CodeSniffer/commit/e85ff8a6dc016743d23211e19061a696ae7aa1d3#diff-eaa28b11ab639fd52744d02ceb5ea6ba

Thanks for the report.
